src-self-hosted/translate_c.zig

```diff
@@ -1784,25 +2342,24 @@ fn transCreateNodePrefixOp(
     return node;
 }
 
-fn transCreateNodeInfixOp(
+fn transCreateNodeInfixOpImpl(
     rp: RestorePoint,
     scope: *Scope,
-    stmt: *const ZigClangBinaryOperator,
+    lhs_node: *ast.Node,
+    rhs_node: *ast.Node,
     op: ast.Node.InfixOp.Op,
     op_tok_id: std.zig.Token.Id,
     bytes: []const u8,
     grouped: bool,
 ) !*ast.Node {
     const lparen = if (grouped) try appendToken(rp.c, .LParen, "(") else undefined;
-    const lhs = try transExpr(rp, scope, ZigClangBinaryOperator_getLHS(stmt), .used, .l_value);
     const op_token = try appendToken(rp.c, op_tok_id, bytes);
-    const rhs = try transExpr(rp, scope, ZigClangBinaryOperator_getRHS(stmt), .used, .r_value);
```

Comment:
Due to how zig fmt works this op_token must be added before any of the rhs tokens. See https://github.com/ziglang/zig/blob/2c7a2aefbfd0dbab190f912b4fbcbda96fb5ac44/lib/std/zig/render.zig#L1633

Reply:
`lparen` must also be added before any of the `lhs` tokens.

Reply:
```zig
fn transCreateNodeInfixOp(
    rp: RestorePoint,
    scope: *Scope,
    lparen: ?ast.TokenIndex,
    lhs_node: *ast.Node,
    op: ast.Node.InfixOp.Op,
    op_tok_id: ast.TokenIndex,
    rhs_node: *ast.Node,
) !*ast.Node {
    const node = try rp.c.a().create(ast.Node.InfixOp);
    node.* = .{
        .op_token = op_tok_id,
        .lhs = lhs_node,
        .op = op,
        .rhs = rhs_node,
    };
    if (lparen == null) return &node.base;
    const rparen = try appendToken(rp.c, .RParen, ")");
    const grouped_expr = try rp.c.a().create(ast.Node.GroupedExpression);
    grouped_expr.* = .{
        .lparen = lparen.?,
        .expr = &node.base,
        .rparen = rparen,
    };
    return &grouped_expr.base;
}
```
It's more verbose at call sites but it gives correct behavior.

Reply:
FYI, if you do end up taking on this task, I think passing in hard-coded parens here is not the right solution. I think it should just parse grouped expression and translate them over as-is, if possible.

I believe I had this working earlier before restarting my work, and I don't think it was that hard to do.

If so, this `transCreateNodeInfixOp` function becomes quite a bit less verbose, so the duplication isn't a big deal at that point.